File: flutter_app/lib/screens/dashboard/dashboard_screen.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import 'package:http/http.dart' as http;
import 'dart:convert';
import '../../constants/colors.dart';
import '../../constants/config.dart';
import '../../providers/auth_provider.dart';
import '../../api/client.dart';

class DashboardScreen extends StatefulWidget {
  const DashboardScreen({super.key});

  @override
  State<DashboardScreen> createState() => _DashboardScreenState();
}

class _DashboardScreenState extends State<DashboardScreen> {
  Map<String, dynamic>? _overview;

  @override
  void initState() {
    super.initState();
    _fetch();
  }

  Future<void> _fetch() async {
    try {
      final token = await ApiClient.getToken();
      if (token == null) return;
      final res = await http.get(
        Uri.parse('${AppConfig.apiBaseUrl}/dashboard/overview'),
        headers: {'Authorization': 'Bearer $token'},
      );
      if (res.statusCode == 200) setState(() => _overview = jsonDecode(res.body));
    } catch (_) {}
  }

  static const _menuItems = [
    {'key': 'orders', 'label': 'My Orders', 'icon': '📋', 'route': '/orders'},
    {'key': 'wallet', 'label': 'Wallet', 'icon': '💰', 'route': '/wallet'},
    {'key': 'subscription', 'label': 'Subscriptions', 'icon': '📅', 'route': '/subscriptions'},
    {'key': 'products', 'label': 'Purchases', 'icon': '📦', 'route': '/products'},
    {'key': 'profile', 'label': 'Profile', 'icon': '👤', 'route': '/profile'},
  ];

  @override
  Widget build(BuildContext context) {
    final auth = context.watch<AuthProvider>();
    final user = auth.user;

    return Scaffold(
      backgroundColor: AppColors.background,
      appBar: AppBar(
        title: const Text('Dashboard', style: TextStyle(color: AppColors.textPrimary, fontWeight: FontWeight.w700)),
        backgroundColor: AppColors.surface,
        iconTheme: const IconThemeData(color: AppColors.textPrimary),
        actions: [
          TextButton(
            onPressed: () { auth.logout(); Navigator.pushReplacementNamed(context, '/login'); },
            child: const Text('Logout', style: TextStyle(color: AppColors.error, fontSize: 12)),
          ),
        ],
      ),
      body: RefreshIndicator(
        color: AppColors.neonCyan,
        onRefresh: _fetch,
        child: ListView(
          padding: const EdgeInsets.all(16),
          children: [
            Container(
              padding: const EdgeInsets.all(16),
              decoration: BoxDecoration(
                color: AppColors.cardBg, borderRadius: BorderRadius.circular(16),
                border: Border.all(color: AppColors.border),
              ),
              child: Row(
                children: [
                  CircleAvatar(
                    radius: 24,
                    backgroundColor: AppColors.electricViolet.withValues(alpha: 0.2),
                    child: Text('${(user?['name'] as String? ?? 'U')[0].toUpperCase()}', style: const TextStyle(fontSize: 20, fontWeight: FontWeight.w700, color: AppColors.electricViolet)),
                  ),
                  const SizedBox(width: 14),
                  Column(
                    crossAxisAlignment: CrossAxisAlignment.start,
                    children: [
                      Text('${user?['name'] ?? 'User'}', style: const TextStyle(fontSize: 16, fontWeight: FontWeight.w700, color: AppColors.textPrimary)),
                      Text('${user?['email'] ?? ''}', style: const TextStyle(fontSize: 12, color: AppColors.textMuted)),
                    ],
                  ),
                ],
              ),
            ),
            const SizedBox(height: 16),
            if (_overview != null)
              Wrap(
                spacing: 8,
                runSpacing: 8,
                children: [
                  _statBox('${_overview!['activeOrders'] ?? 0}', 'Active Orders'),
                  _statBox('₹${_overview!['walletBalance'] ?? 0}', 'Wallet'),
                  _statBox('${_overview!['loyaltyPoints'] ?? 0}', 'Points'),
                  _statBox('₹${_overview!['referralEarnings'] ?? 0}', 'Referrals'),
                ],
              ),
            const SizedBox(height: 16),
            Wrap(
              spacing: 10,
              runSpacing: 10,
              children: _menuItems.map((item) => GestureDetector(
                onTap: () => Navigator.pushNamed(context, item['route'] as String),
                child: Container(
                  width: (MediaQuery.of(context).size.width - 52) / 3,
                  padding: const EdgeInsets.all(16),
                  decoration: BoxDecoration(
                    color: AppColors.cardBg,
                    borderRadius: BorderRadius.circular(14),
                    border: Border.all(color: AppColors.border),
                  ),
                  child: Column(
                    children: [
                      Text('${item['icon']}', style: const TextStyle(fontSize: 28)),
                      const SizedBox(height: 8),
                      Text('${item['label']}', style: const TextStyle(fontSize: 12, color: AppColors.textSecondary), textAlign: TextAlign.center),
                    ],
                  ),
                ),
              )).toList(),
            ),
          ],
        ),
      ),
    );
  }

  Widget _statBox(String value, String label) {
    return Container(
      width: (MediaQuery.of(context).size.width - 52) / 2,
      padding: const EdgeInsets.all(14),
      decoration: BoxDecoration(
        color: AppColors.cardBg,
        borderRadius: BorderRadius.circular(12),
        border: Border.all(color: AppColors.border),
      ),
      child: Column(
        children: [
          Text(value, style: const TextStyle(fontSize: 18, fontWeight: FontWeight.w800, color: AppColors.neonCyan)),
          const SizedBox(height: 4),
          Text(label, style: const TextStyle(fontSize: 11, color: AppColors.textMuted)),
        ],
      ),
    );
  }
}

File: Calcutta Node App/app/src/screens/dashboard/DashboardScreen.js
import React, { useState, useEffect } from 'react';
import { View, Text, ScrollView, TouchableOpacity, StyleSheet, RefreshControl } from 'react-native';
import { LinearGradient } from 'expo-linear-gradient';
import { useAuth } from '../../context/AuthContext';
import api from '../../api/client';
import { colors } from '../../constants/colors';

const menuItems = [
  { key: 'orders', label: 'My Orders', icon: '📋', screen: 'Orders' },
  { key: 'wallet', label: 'Wallet', icon: '💰', screen: 'Wallet' },
  { key: 'subscription', label: 'Subscriptions', icon: '📅', screen: 'Subscriptions' },
  { key: 'products', label: 'Purchases', icon: '📦', screen: 'Products' },
  { key: 'profile', label: 'Profile', icon: '👤', screen: 'Profile' },
];

export default function DashboardScreen({ navigation }) {
  const { user, logout, refreshUser } = useAuth();
  const [overview, setOverview] = useState(null);
  const [refreshing, setRefreshing] = useState(false);

  useEffect(() => {
    fetchOverview();
  }, []);

  async function fetchOverview() {
    try {
      const { data } = await api.get('/dashboard/overview');
      setOverview(data);
    } catch {}
  }

  function onRefresh() {
    setRefreshing(true);
    Promise.all([fetchOverview(), refreshUser()]).finally(() => setRefreshing(false));
  }

  return (
    <ScrollView style={styles.container} refreshControl={<RefreshControl refreshing={refreshing} onRefresh={onRefresh} tintColor={colors.neonCyan} />}>
      <LinearGradient colors={['rgba(0,229,255,0.06)', 'transparent']} style={styles.grad} />
      <View style={styles.header}>
        <Text style={styles.title}>Dashboard</Text>
        <TouchableOpacity onPress={logout} style={styles.logoutBtn}>
          <Text style={styles.logoutText}>Logout</Text>
        </TouchableOpacity>
      </View>

      <View style={styles.profileCard}>
        <View style={styles.avatar}>
          <Text style={styles.avatarText}>{user?.name?.charAt(0) || 'U'}</Text>
        </View>
        <View>
          <Text style={styles.userName}>{user?.name || 'User'}</Text>
          <Text style={styles.userEmail}>{user?.email || ''}</Text>
        </View>
      </View>

      {overview && (
        <View style={styles.statsRow}>
          <View style={styles.statBox}>
            <Text style={styles.statValue}>{overview.activeOrders || 0}</Text>
            <Text style={styles.statLabel}>Active Orders</Text>
          </View>
          <View style={styles.statBox}>
            <Text style={styles.statValue}>₹{overview.walletBalance || 0}</Text>
            <Text style={styles.statLabel}>Wallet</Text>
          </View>
          <View style={styles.statBox}>
            <Text style={styles.statValue}>{overview.loyaltyPoints || 0}</Text>
            <Text style={styles.statLabel}>Points</Text>
          </View>
          <View style={styles.statBox}>
            <Text style={styles.statValue}>₹{overview.referralEarnings || 0}</Text>
            <Text style={styles.statLabel}>Referrals</Text>
          </View>
        </View>
      )}

      <View style={styles.menuGrid}>
        {menuItems.map((item) => (
          <TouchableOpacity key={item.key} style={styles.menuCard} onPress={() => navigation.navigate(item.screen)}>
            <Text style={styles.menuIcon}>{item.icon}</Text>
            <Text style={styles.menuLabel}>{item.label}</Text>
          </TouchableOpacity>
        ))}
      </View>

      <View style={{ height: 100 }} />
    </ScrollView>
  );
}

const styles = StyleSheet.create({
  container: { flex: 1, backgroundColor: colors.background },
  grad: { position: 'absolute', top: 0, left: 0, right: 0, height: 200 },
  header: { flexDirection: 'row', justifyContent: 'space-between', alignItems: 'center', paddingHorizontal: 20, paddingTop: 56, paddingBottom: 16 },
  title: { fontSize: 26, fontWeight: '800', color: colors.textPrimary },
  logoutBtn: { paddingHorizontal: 14, paddingVertical: 6, borderRadius: 8, backgroundColor: colors.error + '20' },
  logoutText: { color: colors.error, fontSize: 12, fontWeight: '600' },
  profileCard: { flexDirection: 'row', alignItems: 'center', marginHorizontal: 16, backgroundColor: colors.cardBg, borderRadius: 16, padding: 16, borderWidth: 1, borderColor: colors.border, marginBottom: 16 },
  avatar: { width: 48, height: 48, borderRadius: 24, backgroundColor: colors.electricViolet + '30', justifyContent: 'center', alignItems: 'center', marginRight: 14 },
  avatarText: { fontSize: 20, fontWeight: '700', color: colors.electricViolet },
  userName: { fontSize: 16, fontWeight: '700', color: colors.textPrimary },
  userEmail: { fontSize: 12, color: colors.textMuted, marginTop: 2 },
  statsRow: { flexDirection: 'row', flexWrap: 'wrap', marginHorizontal: 12, gap: 8, marginBottom: 20 },
  statBox: { flex: 1, minWidth: '45%', backgroundColor: colors.cardBg, borderRadius: 12, padding: 14, borderWidth: 1, borderColor: colors.border, alignItems: 'center' },
  statValue: { fontSize: 18, fontWeight: '800', color: colors.neonCyan },
  statLabel: { fontSize: 11, color: colors.textMuted, marginTop: 4 },
  menuGrid: { flexDirection: 'row', flexWrap: 'wrap', paddingHorizontal: 12, gap: 10 },
  menuCard: { width: '30%', flex: 1, minWidth: 100, backgroundColor: colors.cardBg, borderRadius: 14, padding: 16, borderWidth: 1, borderColor: colors.border, alignItems: 'center' },
  menuIcon: { fontSize: 28, marginBottom: 8 },
  menuLabel: { fontSize: 12, color: colors.textSecondary, fontWeight: '500', textAlign: 'center' },
});
